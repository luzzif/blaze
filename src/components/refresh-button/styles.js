export const styles = {
    root: {
        width: 32,
        height: 32,
        "& svg": {
            fontSize: 20
        }
    }
};
