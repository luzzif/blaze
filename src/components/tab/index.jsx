import React from "react";
import { Card, CardContent } from "@material-ui/core";

export const Tab = props => (
    <Card>
        <CardContent />
    </Card>
);
