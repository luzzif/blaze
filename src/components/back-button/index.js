import React from "react";
import { IconButton, withStyles } from "@material-ui/core";
import { ArrowBack } from "@material-ui/icons";
import { compose } from "redux";
import { styles } from "./styles";
import PropTypes from "prop-types";

const BasicBackButton = ({ classes, ...rest }) => (
    <IconButton {...rest} className={classes.root}>
        <ArrowBack />
    </IconButton>
);

BasicBackButton.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const BackButton = enhanceComponent(BasicBackButton);
