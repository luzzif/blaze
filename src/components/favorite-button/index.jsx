import React from "react";
import { IconButton, withStyles } from "@material-ui/core";
import { Favorite } from "@material-ui/icons";
import { compose } from "redux";
import { styles } from "./styles";
import PropTypes from "prop-types";

const BasicFavoriteButton = ({ classes, ...rest }) => (
    <IconButton {...rest} className={classes.root}>
        <Favorite />
    </IconButton>
);

BasicFavoriteButton.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const FavoriteButton = enhanceComponent(BasicFavoriteButton);
