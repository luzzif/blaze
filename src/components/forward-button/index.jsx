import React from "react";
import { IconButton, withStyles } from "@material-ui/core";
import { ArrowForward } from "@material-ui/icons";
import { compose } from "redux";
import { styles } from "./styles";
import PropTypes from "prop-types";

const BasicForwardButton = ({ classes, ...rest }) => (
    <IconButton {...rest} className={classes.root}>
        <ArrowForward />
    </IconButton>
);

BasicForwardButton.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const ForwardButton = enhanceComponent(BasicForwardButton);
