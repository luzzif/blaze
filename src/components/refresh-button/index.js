import React from "react";
import { IconButton, withStyles } from "@material-ui/core";
import { Refresh } from "@material-ui/icons";
import { compose } from "redux";
import { styles } from "./styles";
import PropTypes from "prop-types";

const BasicRefreshButton = ({ classes, ...rest }) => (
    <IconButton {...rest} className={classes.root}>
        <Refresh />
    </IconButton>
);

BasicRefreshButton.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const RefreshButton = enhanceComponent(BasicRefreshButton);
