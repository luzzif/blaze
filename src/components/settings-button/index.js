import React from "react";
import PropTypes from "prop-types";
import { IconButton, withStyles } from "@material-ui/core";
import { Menu } from "@material-ui/icons";
import { compose } from "redux";
import { styles } from "./styles";

const BasicSettingsButton = ({ classes, ...rest }) => (
    <IconButton {...rest} classes={{ root: classes.root }}>
        <Menu />
    </IconButton>
);

BasicSettingsButton.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const SettingsButton = enhanceComponent(BasicSettingsButton);
