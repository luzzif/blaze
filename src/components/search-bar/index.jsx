import React from "react";
import PropTypes from "prop-types";
import { Card, CardContent, withStyles, Grid } from "@material-ui/core";
import { compose } from "redux";
import { styles } from "./styles";
import { FavoriteButton } from "../favorite-button";

const BasicSearchBar = ({ classes, ...rest }) => (
    <Card {...rest} classes={{ root: classes.root }}>
        <CardContent classes={{ root: classes.content }}>
            <div className={classes.container}>
                <input className={classes.input} type="text" />
                <FavoriteButton />
            </div>
        </CardContent>
    </Card>
);

BasicSearchBar.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const SearchBar = enhanceComponent(BasicSearchBar);
