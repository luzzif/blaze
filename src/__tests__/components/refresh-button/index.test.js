import React from "react";
import { Refresh } from "@material-ui/icons";
import { RefreshButton } from "../../../components/refresh-button";
import TestRenderer from "react-test-renderer";

describe("favorite button", () => {
    let testInstance = null;

    beforeEach(() => {
        testInstance = TestRenderer.create(<RefreshButton />);
    });

    it("renders correctly", () => {
        expect(testInstance.toJSON().type).toBe("button");
        expect(testInstance).toMatchSnapshot();
    });

    it("should be clickable", () => {
        expect(testInstance.root.findByType("button").props.disabled).toBe(false);
        expect(testInstance).toMatchSnapshot();
    });

    it("should contain a refresh icon", () => {
        expect(testInstance.root.findByType(Refresh)).toBeTruthy();
        expect(testInstance).toMatchSnapshot();
    });
});
