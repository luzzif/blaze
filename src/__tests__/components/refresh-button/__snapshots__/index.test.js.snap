exports[`favorite button renders correctly 1`] = `
<button
  className="MuiButtonBase-root-8 MuiIconButton-root-2 BasicRefreshButton-root-1"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button">
  <span
    className="MuiIconButton-label-7">
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-11"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24">
      <path
        d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z" />
      <path
        d="M0 0h24v24H0z"
        fill="none" />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-18" />
</button>
`;

exports[`favorite button should be clickable 1`] = `
<button
  className="MuiButtonBase-root-8 MuiIconButton-root-2 BasicRefreshButton-root-1"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button">
  <span
    className="MuiIconButton-label-7">
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-11"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24">
      <path
        d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z" />
      <path
        d="M0 0h24v24H0z"
        fill="none" />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-18" />
</button>
`;

exports[`favorite button should contain a refresh icon 1`] = `
<button
  className="MuiButtonBase-root-8 MuiIconButton-root-2 BasicRefreshButton-root-1"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button">
  <span
    className="MuiIconButton-label-7">
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-11"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24">
      <path
        d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z" />
      <path
        d="M0 0h24v24H0z"
        fill="none" />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-18" />
</button>
`;
